raiden-ts/docs-source/SDK-Development.md

```diff
@@ -118,6 +118,45 @@ The project is structured in a domain-driven logic, each folder under `src` repr
 
 These are just suggestions on how to keep a well organized codebase. It is the developer's responsibility to decide in which module/domain any function, data or type belongs.
 
+## Data Persistence
+
+The visual representation of the data handling and persistance:
+
+```
+                         +--------------------+
+                         |       State        |
+          .............. |      [Redux]       | <.............
+          .              |    (in memory)     |              .
+          .              +--------------------+              .
+          .                        ^                         .
+          .                        .                         .
+   async sync back         load data on start        read historic data
+          .                        .                         .
+          .                        .                         .
+          .              +--------------------+              .
+          .              |     Database       |              .
+          .............> |     [PouchDB]      | ..............
+                         |(persistent storage)|
+                         +--------------------+
+                                   ^
+                                   |
+          +------------------------+------------------------+
+          |                        |                        |
++--------------------+   +--------------------+   +--------------------+
+|  LevelDB Adapter   |   | IndexedDB Adapter  |   |   Memory Adapter   |
+|     (NodeJS)       |   |   (Web-Browser)    |   |      (Tests)       |
++--------------------+   +--------------------+   +--------------------+
+```
+
+Additional notes:
+
+- The database adapter gets chosen based on the environment the SDK is running inside.
+- The hot state in memory with Redux is limiting its size by not including historic data of transfers and old channels.
+- The data synchronization from the memory to the storage does not block the protocol logic and gets minimized by state diffs.
+- The database is used as source to do dump backups or load such one (can be transferred between clients).
+- Old database scheme versions get automatically migrated during startup before the load to the Redux state.
+- Clients requests which try to read historic data might be slower due to the database read, while all protocol relevant operations are blazing fast, operating on in-memory data.
+
```

Comment:
Maybe it'd be interesting to add here that after all epics complete and all actions go through, we have logic in place to wait for the last state to be synced to the database before closing it and only then the `Raiden.close` promise is resolved. This isn't critical for fast enough implementations, which are expected to always sync the state changes mostly in realtime, but when possible it's interesting to await on this before exiting the process. This is done on CLI, while it's harder to prevent a browser tab from being closed without proper shutdown/logout.

Reply:
For reference: 
- https://github.com/raiden-network/light-client/blob/master/raiden-ts/src/epics.ts#L280-L292
- https://github.com/raiden-network/light-client/blob/master/raiden-ts/src/persister.ts#L102-L106

Reply:
:heavy_check_mark: